nova_vm/src/ecmascript/types/language/string/data.rs

```diff
@@ -22,6 +49,28 @@ impl StringHeapData {
         }
     }
 
+    pub fn utf16_len(&self) -> usize {
+        match &self.mapping {
+            IndexMapping::Ascii => self.len(),
+            IndexMapping::NonAscii { mapping } => mapping.len(),
+        }
+    }
+
+    // TODO: This should return a wtf8::CodePoint.
+    pub fn utf16_char(&self, idx: usize) -> char {
+        let utf8_idx = match &self.mapping {
+            IndexMapping::Ascii => idx,
+            IndexMapping::NonAscii { mapping } => {
+                // TODO: Deal with surrogates.
```

Comment:
**question:** What does this actually give us in JS? Is it just the replacement character?

Reply:
No, in those cases JS would return a string containing a lone surrogate. I didn't do this in this PR because a number of places assume that strings must be UTF-8, and I'll be tackling that in a follow-up.